tests/e2e/IExecDataProtector/fetchProtectedData.test.ts

```diff
@@ -62,4 +62,46 @@ describe('dataProtector.fetchProtectedData()', () => {
       new ValidationError('owner should be an ethereum address')
     );
   });
+  it('pagination: fetches the first 1000 items by default', async () => {
+    const res = await dataProtector.fetchProtectedData();
+    expect(res.length).toBe(1000);
+  });
+  it('pagination: fetches a specific page with a specified page size', async () => {
+    const page = '2'; // Specify the desired page number
+    const pageSize = '50'; // Specify the desired page size
+    const res = await dataProtector.fetchProtectedData({ page, pageSize });
+
+    // Check if the correct number of items for the specified page size is retrieved
+    expect(res.length).toBe(50);
+    // TODO: implement logic to ensure that the items are from the desired page.
```

Comment:
Maybe we can perform a second request with `page: 0, pageSize: 150` and check that two elements are the same?

Looks like it works:

```
  it('pagination: fetches a specific page with a specified page size', async () => {
    const page = 2; // Specify the desired page number
    const pageSize = 50; // Specify the desired page size
    const res = await dataProtector.fetchProtectedData({ page, pageSize });

    // Check if the correct number of items for the specified page size is retrieved
    expect(res.length).toBe(50);

    const res2ToCheck = await dataProtector.fetchProtectedData({
      page: 0,
      pageSize: 150,
    });
    expect(res[49]).toEqual(res2ToCheck[149]);
  });
```